lib/elements/satin_column.py

```diff
@@ -1479,21 +1499,31 @@ def to_stitch_groups(self, last_patch=None):
             patch += self.do_center_walk()
 
         if self.contour_underlay:
-            patch += self.do_contour_underlay()
+            contour = self.do_contour_underlay()
+            if patch.stitches:
+                self.add_running_stitches(patch.stitches[-1], contour.stitches[0], patch)
+            patch += contour
 
         if self.zigzag_underlay:
             # zigzag underlay comes after contour walk underlay, so that the
             # zigzags sit on the contour walk underlay like rail ties on rails.
-            patch += self.do_zigzag_underlay()
+            zigzag = self.do_zigzag_underlay()
+            if patch.stitches:
+                self.add_running_stitches(patch.stitches[-1], zigzag.stitches[0], patch)
```

Comment:
We have this pattern 3 times in this function.  Maybe this should be split out to a function?